uuid.go

```diff
@@ -344,6 +345,30 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+// MarshalJSON marshalls the NullUUID as nil or the nested UUID
+func (u NullUUID) MarshalJSON() ([]byte, error) {
+	if u.Valid == false {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(u.UUID)
+}
+
+// UnmarshalJSON unmarshalls a NullUUID
+func (u *NullUUID) UnmarshalJSON(b []byte) error {
+	var s *UUID
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	if s != nil {
+		u.Valid = true
+		u.UUID = *s
+	} else {
+		u.Valid = false
+	}
+	return nil
+}
```

Comment:
> So, you removed `u.Valid = false` part. I thought it would be better to explicitly initialize all the fields and not rely on default initialization. I don't mind changing that if you want.

I think you _need_ to explicitly initialize all fields, you cannot rely on default initialization. The value being unmarshaled into may be zero, or it may not be zero. Removing `u.Valid = false` would make it fail to operate correctly on `&NullUUID{Valid: true}` starting value.